### PR TITLE
feat: Add (partial) archive node support

### PR DESCRIPTION
**Pull request by Yawning**
* [x] Add a composite client
 * [x] Wire in the composite client

### CODE REVIEW

**Comment by Yawning** [comment]
This maybe is enough to get historical EVM queries supported, but I have no idea if this actually works.  Someone that cares should test this, because I do not.

File: client/multi.go [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("client: failed to query active rt info: %w", err)
+	}
+
+	mc := &MultiClient{
+		active:         active,
+		archives:       make([]*archiveClient, 0, len(archives)),
+		archivesEldest: math.MaxUint64,
+	}
+	if mc.archivesEldest, _, err = clientBlockRanges(ctx, active); err != nil {
```
Comment (ptrus) [inline client/multi.go]: This should set `activeEldest` right? (not archive)

Comment (Yawning) [reply client/multi.go]: yes

**Review by ptrus — APPROVED** [review]

**Comment by Yawning** [comment]
Welp, this doesn't actually work at all because the client-sdk can only talk to the version of the oasis-node it was built against.  The ideal solution is to wait for someone (that's not me) to implement https://github.com/oasisprotocol/oasis-sdk/issues/954.

**Comment by gw0** [comment]
An alternative solution is to import older versions of `client-sdk` by changing the Git versioning or by using the `replace` directive (https://www.percona.com/blog/2020/03/09/using-different-versions-of-a-package-in-an-application-via-go-modules/). Then for each archive node a different version should be used (either explicitly provided in the config or implicitly by querying the supported versions).

**Comment by Yawning** [comment]
> An alternative solution is to import older versions of `client-sdk` by changing the Git versioning or by using the `replace` directive (https://www.percona.com/blog/2020/03/09/using-different-versions-of-a-package-in-an-application-via-go-modules/). Then for each archive node a different version should be used (either explicitly provided in the config or implicitly by querying the supported versions).

That is an objectively terrible solution because it relies on the public interfaces exported by the client sdk not changing at all between versions, or a mountain of code duplication in this codebase.  This doesn't actually work (easily anyway) either because `replace` isn't very good, and you get things like:

```
# github.com/oasisprotocol/oasis-core/go/v21/common/grpc
../../../go/pkg/mod/github.com/oasisprotocol/oasis-core/go@v0.2103.13/common/grpc/cbor.go:20:14: undefined: cbor.UnmarshalHigherLimits
```

**Comment by Yawning** [comment]
Unless someone has better ideas, this is a total non-starter (even having the SDK magically handle backward compatibility is.... "hard if not way more trouble than it's worth"), so I am tempted to close this PR as a WONTFIX.

**Comment by Yawning** [comment]
This approach has way too many problems, and making it work is 100% more trouble than it's worth.

**Pull request closed without merging**